assets/javascripts/test_result.js

```diff
@@ -811,22 +822,71 @@ function renderInvestigationTab(response) {
         valueElement.appendChild(html);
       }
     } else {
-      var textLines = typeof value === 'string' ? value.split('\n') : [value];
-      var textLinesRest;
-
-      var lineLimit = 10;
-      if (textLines.length > lineLimit) {
-        textLinesRest = textLines.slice(lineLimit, textLines.length);
-        textLines = textLines.slice(0, lineLimit);
+      var preElement = document.createElement('pre');
+      var preElementMore = document.createElement('pre');
+      var enable_more = false;
+      if (key === 'error') {
+        preElement.appendChild(document.createTextNode(value));
+      }
+      if (['test_log', 'needles_log'].indexOf(key) >= 0) {
+        var repoUrl = getInvestigationDataAttr(key);
+        if (repoUrl) {
+          var gitstats = githashToLink(value, repoUrl);
+          // assume string 'No test changes..'
+          if (gitstats == null) {
+            preElement.appendChild(document.createTextNode(value));
+          } else {
+            for (let i = 0; i < gitstats.length; i++) {
+              var statItem = document.createElement('div');
+              var collapseSign = document.createElement('a');
+              collapseSign.className = 'collapsed';
+              collapseSign.setAttribute('href', '#collapseEntry' + i);
+              collapseSign.setAttribute('data-toggle', 'collapse');
+              collapseSign.setAttribute('aria-expanded', 'false');
+              collapseSign.setAttribute('aria-controls', 'collapseEntry');
+              collapseSign.innerHTML = '+ ';
+              collapseSign.setAttribute('onclick', 'toggleSign(this)');
+              var spanElem = document.createElement('span');
+              var logDetailsDiv = document.createElement('div');
+              logDetailsDiv.id = 'collapseEntry' + i;
+              logDetailsDiv.className = 'collapse';
+              stats = gitstats[i].split('\n')[0];
+              spanElem.innerHTML = stats;
+              logDetailsDiv.innerHTML = gitstats[i]
+                .split('\n')
+                .slice(1, gitstats.length - 1)
+                .join('\n');
+              statItem.append(collapseSign, spanElem, logDetailsDiv);
+
+              if (i < 5) {
+                preElement.appendChild(statItem);
+              } else {
+                enable_more = true;
+                preElementMore.appendChild(statItem);
+              }
+            }
+          }
+        } else {
+          var textLines = typeof value === 'string' ? value.split('\n') : [value];
+          var textLinesRest;
+
+          var lineLimit = 10;
```

Comment:
same as above, magic number. Please define a constant with explanatory name

Reply:
*this change is ready for review*